File: CODIGO 10/work.rb
require 'mathn'

def cloud_altitude(d1, a, b)
  h = tan(b) * d1 / (tan(a) - tan(b))
  h.round
end

n = gets.to_i

heights = []
for i in 0..n-1
  d1, a, b = gets.split.map(&:to_f)
  heights.push(cloud_altitude(d1, a, b))
end

heights.each { |h| puts h }

File: CODIGO 10/work.pl
use strict;
use warnings;

sub cloud_altitude {
  my ($d1, $a, $b) = @_;

  my $h = tan($b) * $d1 / (tan($a) - tan($b));

  return int($h);
}

my $n = <>;

my @heights;
for (1..$n) {
  my @data = split ' ', <>;

  my $d1 = $data[0];
  my $a = $data[1];
  my $b = $data[2];

  push @heights, cloud_altitude($d1, $a, $b);
}

print "@heights\n";
